en/02_Developer_Guides/00_Model/04_Data_Types_and_Casting.md

```diff
@@ -31,33 +31,41 @@ class Player extends DataObject
+| `Date` | A date field | Must be a valid date in `Y-m-d` format | [`DBDate`](api:SilverStripe\ORM\FieldType\DBDate) |
+| `Datetime` | A date/time field | Must be a valid datetime in `Y-m-d H:i:s` format | [`DBDatetime`](api:SilverStripe\ORM\FieldType\DBDatetime) |
+| `DBClassName` | A special enumeration for storing class names | Must be a valid FQCN of a `DataObject` subclass | [`DBClassName`](api:SilverStripe\ORM\FieldType\DBClassName) |
+| `DBClassNameVarchar` | A special enumeration for storing class names in a `Varchar` field | Must be a valid FQCN of a `DataObject` subclass | [`DBClassNameVarchar`](api:SilverStripe\ORM\FieldType\DBClassNameVarchar) |
+| `Decimal` | A decimal number | Must be a decimal. | [`DBDecimal`](api:SilverStripe\ORM\FieldType\DBDecimal) |
+| `Double` | A floating point number with double precision | Must be numeric | [`DBDouble`](api:SilverStripe\ORM\FieldType\DBDouble) |
+| `Email` | An email field | Must be a valid email address | [`DBEmail`](api:SilverStripe\ORM\FieldType\DBEmail) |
+| `Enum` | An enumeration of a set of strings that can store a single value | Must be one of the defined values | [`DBEnum`](api:SilverStripe\ORM\FieldType\DBEnum) |
+| `Float` | A floating point number | Must be numeric | [`DBFloat`](api:SilverStripe\ORM\FieldType\DBFloat) |
+| `ForeignKey` | A special `Int` field used for foreign keys in `has_one` relationships | Must be an int | [`DBForeignKey`](api:SilverStripe\ORM\FieldType\DBForeignKey) |
+| `HTMLFragment` | A variable-length string of up to 2MB, designed to store HTML. Doesn't process [shortcodes](/developer_guides/extending/shortcodes/) | Must be a string | [`DBHTMLText`](api:SilverStripe\ORM\FieldType\DBHTMLText) |
+| `HTMLText` | A variable-length string of up to 2MB, designed to store HTML. Processes [shortcodes](/developer_guides/extending/shortcodes/) | Must be a string | [`DBHTMLText`](api:SilverStripe\ORM\FieldType\DBHTMLText) |
+| `HTMLVarchar` | A variable-length string of up to 255 characters, designed to store HTML. Can process [shortcodes](/developer_guides/extending/shortcodes/) with additional configuration | String must not be longer than specified length | [`DBHTMLVarchar`](api:SilverStripe\ORM\FieldType\DBHTMLVarchar) |
+| `Int` | A 32-bit signed integer field | Must be an int between -2147483648 and 2147483647 | [`DBInt`](api:SilverStripe\ORM\FieldType\DBInt) |
+| `IP` | An IP field | Must be a valid IP address, either IPv4 or IPv6 | [`DBIp`](api:SilverStripe\ORM\FieldType\DBIp) |
+| `Locale` | A field for storing locales | Must be a valid locale | [`DBLocale`](api:SilverStripe\ORM\FieldType\DBLocale) |
+| `Money` | Similar to Currency, but with localisation support | Currency string must not be greater than 4 characters, amount must be a decimal | [`DBMoney`](api:SilverStripe\ORM\FieldType\DBMoney) |
```

Comment:
Isn't it 4 digits including the numbers after the decimal place? e.g. `12.34` is 5 characters. Looking at the framework PR, there's nothing specific for `DBMoney` so why does it say here it has specific validation? Probably just update to use the same wording used for its parent class.

Sorry I didn't notice this before.

Reply:
Funny, I had it wrong it's actually 3 characters

```php
class DBMoney extends DBComposite
{
    private static array $composite_db = [
        'Currency' => 'Varchar(3)',
        'Amount' => 'Decimal(19,4)'
    ];
```
Currency refers to https://en.wikipedia.org/wiki/ISO_4217 - will update docs